snapcraft/plugins/nodejs.py

```diff
@@ -85,8 +94,11 @@ def build(self):
         self._nodejs_tar.provision(self.installdir)
         for pkg in self.options.node_packages:
             self.run(['npm', 'install', '-g', pkg])
-        if os.path.exists(os.path.join(self.builddir, 'package.json')):
-            self.run(['npm', 'install', '-g'])
+        if os.path.exists(os.path.join(self.builddir,
+                                       self.options.node_folder,
+                                       'package.json')):
+            self.run(['npm', 'install', '-g',
+                     os.path.join(self.builddir, self.options.node_folder)])
```

Comment:
Just to verify (not a nodejs expert here): it's safe to do this even when the `package.json` IS in the root of the project?